SimTKcommon/Simulation/include/SimTKcommon/internal/State.h

```diff
@@ -1122,35 +1117,56 @@ was created. This has no effect on the realization level.
 **/
 inline void setSystemTopologyStageVersion(StageVersion topoVersion);
 
-/** (Advanced) Return a StageVersion for time, meaning an integer that is
-incremented whenever time is changed. Will be 1 or greater if time has been
-initialized; otherwise, time is meaningless and may be NaN. **/
-inline StageVersion getTimeStageVersion() const;
 /** (Advanced) Return a StageVersion for q, meaning an integer that is
 incremented whenever any q is changed (or more precisely whenever q or y is
 returned with writable access). Will be 1 or greater if q has been
 allocated. **/
-inline StageVersion getQStageVersion() const;
+inline StageVersion getQValueVersion() const;
 /** (Advanced) Return a StageVersion for u, meaning an integer that is
 incremented whenever any u is changed (or more precisely whenever u or y is
 returned with writable access). Will be 1 or greater if u has been
 allocated. **/
-inline StageVersion getUStageVersion() const;
+inline StageVersion getUValueVersion() const;
 /** (Advanced) Return a StageVersion for z, meaning an integer that is
 incremented whenever any z is changed (or more precisely whenever z or y is
 returned with writable access). Will be 1 or greater if z has been
 allocated. **/
-inline StageVersion getZStageVersion() const;
-
-/** (Advanced) Return a reference to the cache entry information for a 
+inline StageVersion getZValueVersion() const;
```

Comment:
I think "generation" connotes a heavy expensive thing. I like introducing ValueVersion and pretending it's different.